src/sinks/util/http.rs

```diff
@@ -382,11 +382,16 @@ where
         let mut http_client = self.inner.clone();
 
         Box::pin(async move {
-            let request = request_builder(body).await?;
+            let request = request_builder(body).await.map_err(|error| {
+                emit!(SinkRequestBuildError { error: &error });
+                error
+            })?;
             let byte_size = request.body().len();
             let request = request.map(Body::from);
             let (protocol, endpoint) = uri::protocol_endpoint(request.uri().clone());
 
+            // Any errors raised in `http_client.call` results in a `GotHttpWarning` event being emmited
```

Comment:
🤔 unrelated to this change, but since you mentioned it - should that always be a warning or does it already depend on the error raised?

Reply:
Yeah I think it is badly named. Will fix that in a separate PR.